packages/test-utils/src/reporter.ts

```diff
@@ -977,116 +988,121 @@ class ReplayReporter<TRecordingMetadata extends UnstructuredMetadata = Unstructu
   }
 
   async onEnd(): Promise<PendingWork[]> {
-    debug("onEnd");
+    try {
+      debug("onEnd");
+      await this._cacheUserIdPromise;
```

Comment:
I'm not sure but do we want to throw/stop execution is this fails for some reason?

Reply:
I'm not sure. One known error path exists when the api key is invalid. In that case, uploading would be impossible, and we would want to know about that, but I'm not sure if this is the right place to loudly fail as opposed to loudly failing when the upload is attempted.

Reply:
Right now `_cacheUserIdPromise` is only a dependency to set up the logger. If we can't set up the logger, I think we shouldn't fail. Even though for the case where an API key is invalid, it should already be handled somewhere to make that error message in the plugin.

Reply:
That's reasonable.

Reply:
Following up on this: Can we add a catch just for this statement?